packages/smooth_app/lib/pages/preferences/country_selector/country_selector_provider.dart

```diff
@@ -47,146 +48,51 @@ class _CountrySelectorProvider extends PreferencesSelectorProvider<Country> {
-      oFFIsoCodeToCountry[c.offTag.toLowerCase()] = c;
-    }
-    for (final Country c in localizedCountries) {
-      localizedIsoCodeToCountry.putIfAbsent(
-          c.countryCode.toLowerCase(), () => c);
-    }
-    for (final String countryCode in oFFIsoCodeToCountry.keys) {
-      final Country? localizedCountry = localizedIsoCodeToCountry[countryCode];
-      if (localizedCountry == null) {
-        // No localization for the country name was found, use English name as
-        // default.
-        String countryName = oFFIsoCodeToCountry[countryCode]
-            .toString()
-            .replaceAll('OpenFoodFactsCountry.', '')
-            .replaceAll('_', ' ');
-        countryName =
-            '${countryName[0].toUpperCase()}${countryName.substring(1).toLowerCase()}';
-        finalCountriesList.add(
-          Country(
-              name: _fixCountryName(countryName),
-              countryCode: _fixCountryCode(countryCode)),
-        );
-        continue;
-      }
-      final String fixedCountryCode = _fixCountryCode(countryCode);
-      final Country country = fixedCountryCode == countryCode
-          ? localizedCountry
-          : Country(name: localizedCountry.name, countryCode: countryCode);
-      finalCountriesList.add(country);
-    }
-
-    return finalCountriesList;
-  }
+  static List<OpenFoodFactsCountry> _sanitizeCountriesList(String? locale) {
+    final List<OpenFoodFactsCountry> countries =
+        List<OpenFoodFactsCountry>.from(OpenFoodFactsCountry.values);
 
-  /// Fix the countryCode if needed so Backend can process it.
-  static String _fixCountryCode(String countryCode) {
-    // 'gb' is handled as 'uk' in the backend.
-    if (countryCode == 'gb') {
-      countryCode = 'uk';
-    }
-    return countryCode;
-  }
+    countries.sort(
+      (a, b) => (a.getLocalizedName(locale ?? 'en') ?? '')
+          .compareTo(b.getLocalizedName(locale ?? 'en') ?? ''),
+    );
 
-  /// Fix the issues where United Kingdom appears with lowercase 'k'.
-  static String _fixCountryName(String countryName) {
-    if (countryName == 'United kingdom') {
-      countryName = 'United Kingdom';
-    }
-    return countryName;
+    return countries;
   }
 
-  /// Reorder countries alphabetically, bring user's locale country to top.
   static void _reorderCountries(
-    List<Country> countries,
+    List<OpenFoodFactsCountry> countries,
     String? userCountryCode,
   ) {
     countries.sort(
-      (final Country a, final Country b) {
-        if (a.countryCode == userCountryCode) {
-          return -1;
-        }
-        if (b.countryCode == userCountryCode) {
-          return 1;
-        }
-        return a.name.compareTo(b.name);
+      (a, b) {
+        if (a.offTag == userCountryCode) return -1;
+        if (b.offTag == userCountryCode) return 1;
+        return (a.getLocalizedName('en') ?? '')
+            .compareTo(b.getLocalizedName('en') ?? '');
```

Comment:
Please NEVER use magic values like `'en'`.
OpenFoodFactsLanguage.ENGLISH would be more appropriate.